src/api/link_post.js

```diff
@@ -7,6 +24,31 @@ const isValidDID = did => {
   return isIPFS.cid(parts[2])
 }
 
+const isValidAccountType = type => {
+  return Object.values(ACCOUNT_TYPES).includes(type)
+}
+
+const isSupportedChainId = chainId => {
+  return chainId && !!SUPPORTED_CHAINS[chainId]
+}
+
+const isValidTimestamp = timestamp => {
+  return parseInt(timestamp, 10) > 0 && parseInt(timestamp, 10) <= +new Date()
+}
+
+const isValidSignature = async (contractAddress, chainId, sig, msg) => {
+  const abi = [
+    'function isValidSignature(bytes _messageHash, bytes _signature) public view returns (bytes4 magicValue)'
+  ]
+  const network = SUPPORTED_CHAINS[chainId]
+  const ethersProvider = ethers.getDefaultProvider(network)
```

Comment:
What provider will this use? Will that provider always support `eth_call`?

Reply:
according to ethers.js doc: `this creates a FallbackProvider backed by multiple backends (INFURA and Etherscan).`

Reply:
Ok, do you know if Etherscan supports `eth_call` if it doesn't this could be a problem I guess. I think we should probably use infura explicitly and provide a projectID

Reply:
Yes it does, otherwise, it wouldn't be used as a fallback provide.
Here is a call example: https://api.etherscan.io/api?module=proxy&action=eth_call&to=0x4CF488387F035FF08c371515562CBa712f9015d4&data=0x5c975abb